var/spack/repos/builtin/packages/mysql/package.py

```diff
@@ -12,27 +12,106 @@ class Mysql(CMakePackage):
+    version('5.7.18', sha256='0b5d71ed608656cd8181d3bb0434d3e36bac192899038dbdddf5a7594aaea1a2')
+    version('5.7.17', sha256='cebf23e858aee11e354c57d30de7a079754bdc2ef85eb684782458332a4b9651')
+    version('5.7.16', sha256='4935b59974edb275629f6724a0fcf72265a5845faf1e30eeb50ed4b6528318a5')
+    version('5.7.15', sha256='9085353143bfda59c90aa959e79a35622a22aa592e710993416e193b37eb9956')
+    version('5.7.14', sha256='f7415bdac2ca8bbccd77d4f22d8a0bdd7280b065bd646a71a506b77c7a8bd169')
+    version('5.7.13', sha256='50bf1a1635a61235fc43fd4876df2f77163de109372679e29c1ff8dbc38a0b87')
+    version('5.7.12', sha256='32843cb6d22ab22cd2340262b53c0d6009b5bd41b1fa4102beda19635a5c1c87')
+    version('5.7.11', sha256='54f8c7af87d3d8084419bde2b9f0d8970b3dada0757b015981b02f35a3681f0e')
+    version('5.7.10', sha256='1ea1644884d086a23eafd8ccb04d517fbd43da3a6a06036f23c5c3a111e25c74')
+    version('5.7.9',  sha256='315342f5bee1179548cecad2d776cd7758092fd2854024e60a3a5007feba34e0')
+    version('5.6.43', sha256='1c95800bf0e1b7a19a37d37fbc5023af85c6bc0b41532433b3a886263a1673ef')
+    version('5.5.62', sha256='b1e7853bc1f04aabf6771e0ad947f35ac8d237f4b35d0706d1095c9526ff99d7')
+
+    variant('client_only', default=False,
+            description='Build and install client only.')
+    variant('cxxstd',
+            default='default',
+            values=('default', '98', '11', '14', '17'),
+            multi=False,
+            description='Use the specified C++ standard when building.')
+
+    conflicts('+client_only', when='@5.7.0:5.7.999')
+
+    provides('mysql-client')
 
     # https://dev.mysql.com/doc/refman/8.0/en/source-installation.html
 
     # See CMAKE_MINIMUM_REQUIRED in CMakeLists.txt
-    depends_on('cmake@3.8.0:', type='build', when='platform=win32')
-    depends_on('cmake@3.9.2:', type='build', when='platform=darwin')
-    depends_on('cmake@3.4.0:', type='build', when='platform=solaris')
-    depends_on('cmake@2.8.12:', type='build')
+    depends_on('cmake@3.1.0:', type='build', when='@5.7.0:5.7.999 platform=win32')
+    depends_on('cmake@3.8.0:', type='build', when='@8.0.0: platform=win32')
+    depends_on('cmake@3.9.2:', type='build', when='@8.0.0: platform=darwin')
+    depends_on('cmake@3.4.0:', type='build', when='@8.0.0: platform=solaris')
+    depends_on('cmake@2.6:', type='build', when='@:5.6.999')
+    depends_on('cmake@2.8.9:', type='build', when='@5.7.0:5.7.999')
+    depends_on('cmake@2.8.12:', type='build', when='@8.0.0:')
 
     depends_on('gmake@3.75:', type='build')
 
     # Each version of MySQL requires a specific version of boost
     # See BOOST_PACKAGE_NAME in cmake/boost.cmake
-    depends_on('boost@1.68.0', type='build', when='@8.0.15')
-    depends_on('boost@1.66.0', type='build', when='@8.0.11')
-    depends_on('boost@1.59.0', type='build', when='@5.7.22')
+    # 8.0.14+
+    depends_on('boost@1.68.0 cxxstd=default', type='build', when='@8.0.14: cxxstd=default')
```

Comment:
You also wrote https://github.com/spack/spack/pull/10910, which appears to remove the `default` option for boost's `cxxstd` variant.

Reply:
Latest commit accommodates #10910.